lib/promise_utils.js

```diff
@@ -67,7 +67,7 @@ function onceWithCleanup (emitter, event, { timeout = 0, checkCondition = undefi
     })
   }
 
-  task.promise.finally(() => emitter.removeListener(event, onEvent))
+  task.promise.finally(() => emitter.removeListener(event, onEvent)).catch((err) => {})
```

Comment:
This does not seem right. You're silently letting this fail.

Reply:
The promise returned by finally is a copy off the promise the task itself will return. This is why there is a unhandled promise rejection in the first place. No one is handling the promise returned by finally because it is also returned by the promise in task.

Reply:
Altho this does not work with the linter. Maybe `promise.catch(() => {}).finally(() => ...)` would be better?

Reply:
You’re right, I missed that. It’s a chained promise, so it should be catching error removing the event (if I’m not mistaken)